js/templates/serverConnectModal.html

```diff
@@ -0,0 +1,168 @@
+        <% } %>
+      </div>
+        <form class="width100" name="serverConfigForm">
+          <div class="flexCol-12 flex-border borderBottom">
+            <div class="flexRow">
+              <% if (ob.errors.server_ip) { %>
+              <div class="flexCol-12">
+                <ul class="padding2015">
+                  <% ob.errors.server_ip.forEach(function(err) {
+                       print('<li class="list-style-none">' + err + '</li>');
+                     }); %>
+                </ul>
+              </div>
+              <% } %>
+
+              <div class="flexCol-3 borderRight custCol-border">
+                <div class="fieldItem">
+                  <label for="fld_server"><%= polyglot.t('serverConnectModal.serverIP') %></label>
+                </div>
+              </div>
+
+              <div class="flexCol-9 custCol-border <% ob.errors.server_ip && print('invalid') %>">
+                <input name="server_ip" type="text" id="fld_server" placeholder="" class="fieldItem" required="" value="<%= ob.server_ip %>">
+              </div>
+            </div>
+
+            <div class="flexRow">
+              <% if (ob.errors.rest_api_port) { %>
+              <div class="flexCol-12">
+                <ul class="padding2015">
+                  <% ob.errors.rest_api_port.forEach(function(err) {
+                       print('<li class="list-style-none">' + err + '</li>');
+                     }); %>
+                </ul>
+              </div>
+              <% } %>
+
+              <div class="flexCol-3 borderRight custCol-border">
+                <div class="fieldItem">
+                  <label for="fld_rest_port"><%= polyglot.t('serverConnectModal.restApiPort') %></label>
+                </div>
+              </div>
+
+              <div class="flexCol-9 custCol-border <% ob.errors.rest_api_port && print('invalid') %>">
+                <input name="rest_api_port" type="text" id="fld_rest_port" placeholder="" class="fieldItem" value="<%= ob.rest_api_port %>">
+              </div>
+            </div>
+
+            <div class="flexRow">
+              <% if (ob.errors.socket_port) { %>
+              <div class="flexCol-12">
+                <ul class="padding2015">
+                  <% ob.errors.socket_port.forEach(function(err) {
+                       print('<li class="list-style-none">' + err + '</li>');
+                     }); %>
+                </ul>
+              </div>
+              <% } %>
+
+              <div class="flexCol-3 borderRight custCol-border">
+                <div class="fieldItem">
+                  <label for="fld_socket_port"><%= polyglot.t('serverConnectModal.websocketApiPort') %></label>
+                </div>
+              </div>
+
+              <div class="flexCol-9 custCol-border <% ob.errors.socket_port && print('invalid') %>">
+                <input name="socket_port" type="text" id="fld_socket_port" placeholder="" class="fieldItem" value="<%= ob.socket_port %>" disabled>
```

Comment:
Not visible unless you scroll horizontally here, but I'm disabling the socket port for now. I do actually pull the socket port from the server config model which this form saves to. Outside of that, I haven't done a whole lot of work to change the socket failed connection experience. The reason, I disable the field is because once you put in a value to a dead server, you won't have an easy way to change it. Basically, you would need to kill your rest server, so that the Server Connect Modal appears.